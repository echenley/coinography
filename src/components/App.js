// @flow

import React from 'react'
import Expo from 'expo'
import { StackNavigator } from 'react-navigation'
import { Provider } from 'react-redux'
import HomeScreen from './screens/HomeScreen'
import AboutScreen from './screens/AboutScreen'
import store from '../store'

const AppStack = StackNavigator(
  {
    Home: {
      screen: HomeScreen,
    },
    About: {
      screen: AboutScreen,
    },
  },
  {
    initialRouteName: 'Home',
  }
)

type State = {
  loading: boolean,
}

class App extends React.Component<{}, State> {
  state = { loading: true }

  async componentDidMount() {
    await Expo.Font.loadAsync({
      Roboto: require('native-base/Fonts/Roboto.ttf'),
      Roboto_medium: require('native-base/Fonts/Roboto_medium.ttf'),
    })

    this.setState({ loading: false })
  }

  render() {
    if (this.state.loading) {
      return <Expo.AppLoading />
    }

    return (
      <Provider store={store}>
        <AppStack />
      </Provider>
    )
  }
}

export default App
