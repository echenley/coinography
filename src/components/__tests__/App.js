jest.mock('react-navigation', () => {
  const AppStack = () => <div />
  return {
    StackNavigator: () => AppStack,
  }
})
jest.mock('native-base/Fonts/Roboto.ttf', () => 'Roboto')
jest.mock('native-base/Fonts/Roboto_medium.ttf', () => 'Roboto_medium')

import * as React from 'react'
import Expo from 'expo'
import { shallow } from 'enzyme'
import App from '../App'

function renderComponent() {
  return shallow(<App />, { disableLifecycleMethods: true })
}

describe('<App />', () => {
  it('renders correctly while loading', () => {
    expect(renderComponent().getElement()).toMatchSnapshot()
  })

  it('renders correctly after fonts are loaded', async () => {
    const component = renderComponent()
    await component.instance().componentDidMount()
    component.update()
    expect(Expo.Font.loadAsync).toHaveBeenCalledWith({
      Roboto: 'Roboto',
      Roboto_medium: 'Roboto_medium',
    })
    expect(component.getElement()).toMatchSnapshot()
  })
})
