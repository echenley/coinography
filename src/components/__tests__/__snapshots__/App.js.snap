// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly after fonts are loaded 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <AppStack />
</Provider>
`;

exports[`<App /> renders correctly while loading 1`] = `<AppLoading />`;
